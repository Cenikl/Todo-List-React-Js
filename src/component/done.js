import React from "react";

export const Done = (props) => {
const{finished} = props;

    return(<div data-testid="done-1" className="done">
    <h1>Done</h1>
    <hr></hr>
    <ul>
      {finished.map(e => {
        return(
          <li key={e.title} className="listing">
            <p> <h4> Title : </h4> {e.title}</p>
          </li>
        )
      })}
    </ul>
  </div>)
}