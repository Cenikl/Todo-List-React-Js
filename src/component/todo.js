import React, { useState } from 'react';

export const Todo = (props) => {
  const {items,setItems,finished,setFinished,newTitle,setNewTitle} = props;

  const addItem = (event) => {
    if(event.key === 'Enter'){
    if(!newTitle){
      alert("Enter a todo title");
      return;
    }

    const item = {
      title: newTitle
    }; 
    setItems(oldList => [...oldList,item]);
    setNewTitle("");
  }
  }

  const finishWork = (title) => {
    const newFinished = items.filter(item => item.title === title);
    const newArray = items.filter(item => item.title !== title);
    setItems(newArray);
    const finishedItem = {
      title: newFinished[0].title
    }
    setFinished(oldList => [...oldList,finishedItem]);
  }

  
  

  return(
    <div data-testid="todo-1" className="todo">
        <h1>Todo List App</h1>
        <hr></hr>
        <div className="inputing">
        <input type={'text'} 
          placeholder="Put a Title here..." 
          value={newTitle}
          onChange= {e => setNewTitle(e.target.value)}
          onKeyDown={addItem}
          className="titre"/>
          <br/>
        </div>
        <ul>
          {items.map(item => {
            return(
              <li key={item.title} className="listing">
                <p> <h4> Title : </h4> {item.title}</p>
                <input type={'checkbox'} onClick={() => {finishWork(item.title)}}/>
              </li>
            )
          })}
        </ul>
      </div>
  )
}
