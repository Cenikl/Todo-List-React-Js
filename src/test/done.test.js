import React from "react";
import { Done } from "../component/done";
import {render,screen,cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(() =>{
    cleanup();
})

it("renders without crashing",()=>{
    const finished = [];
    render(<Done finished={finished}></Done>)
    const doneElement = screen.getByTestId("done-1");
    expect(doneElement).toBeInTheDocument();
})

it("renders one element",()=>{
    const finished = [{title:"testing"}];
    render(<Done finished={finished}></Done>)
    const doneElement = screen.getByTestId("done-1");
    expect(doneElement).toBeInTheDocument();
    expect(doneElement).toHaveTextContent("testing");
})
it("renders two elements",()=>{
    const finished = [{title:"testing"},{title:"testingtwo"}];
    render(<Done finished={finished}></Done>)
    const doneElement = screen.getByTestId("done-1");
    expect(doneElement).toBeInTheDocument();
    expect(doneElement).toHaveTextContent("testing");
    expect(doneElement).toHaveTextContent("testingtwo");
})
 
it("matches snapshot 1",()=>{
    const finished = [];
    const tree = renderer.create(<Done finished={finished}></Done>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("matches snapshot 2",()=>{
    const finished = [{title:"testing"}];
    const tree = renderer.create(<Done finished={finished}></Done>).toJSON();
    expect(tree).toMatchSnapshot();
})
it("matches snapshot 3",()=>{
    const finished = [{title:"testing"},{title:"testingtwo"}];
    const tree = renderer.create(<Done finished={finished}></Done>).toJSON();
    expect(tree).toMatchSnapshot();
})
