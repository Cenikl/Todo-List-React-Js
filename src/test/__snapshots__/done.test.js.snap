// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="done"
  data-testid="done-1"
>
  <h1>
    Done
  </h1>
  <hr />
  <ul />
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="done"
  data-testid="done-1"
>
  <h1>
    Done
  </h1>
  <hr />
  <ul>
    <li
      className="listing"
    >
      <p>
         
        <h4>
           Title : 
        </h4>
         
        testing
      </p>
    </li>
  </ul>
</div>
`;

exports[`matches snapshot 3 1`] = `
<div
  className="done"
  data-testid="done-1"
>
  <h1>
    Done
  </h1>
  <hr />
  <ul>
    <li
      className="listing"
    >
      <p>
         
        <h4>
           Title : 
        </h4>
         
        testing
      </p>
    </li>
    <li
      className="listing"
    >
      <p>
         
        <h4>
           Title : 
        </h4>
         
        testingtwo
      </p>
    </li>
  </ul>
</div>
`;
